EventFilter/Phase2TrackerRawToDigi/plugins/Phase2TrackerDigiProducer.cc

```diff
@@ -193,46 +204,40 @@ namespace Phase2Tracker {
 
               // store beginning and end of this digis for this detid and add this registry to the list
               // and store data
-              Registry regItemTop(detid + 1, STRIPS_PER_CBC * icbc / 2, proc_work_digis_.size(), stripsTop.size());
-              proc_work_registry_.push_back(regItemTop);
-              proc_work_digis_.insert(proc_work_digis_.end(), stripsTop.begin(), stripsTop.end());
-              Registry regItemBottom(
-                  detid + 2, STRIPS_PER_CBC * icbc / 2, proc_work_digis_.size(), stripsBottom.size());
-              proc_work_registry_.push_back(regItemBottom);
-              proc_work_digis_.insert(proc_work_digis_.end(), stripsBottom.begin(), stripsBottom.end());
+              proc_work_registry.emplace_back(
+                  detid + 1, STRIPS_PER_CBC * icbc / 2, proc_work_digis.size(), stripsTop.size());
+              proc_work_digis.insert(proc_work_digis.end(), stripsTop.begin(), stripsTop.end());
+              proc_work_registry.emplace_back(
+                  detid + 2, STRIPS_PER_CBC * icbc / 2, proc_work_digis.size(), stripsBottom.size());
+              proc_work_digis.insert(proc_work_digis.end(), stripsBottom.begin(), stripsBottom.end());
             }
             ichan++;
           }
         }  // end loop on channels
-        // store digis in edm collections
-        std::sort(proc_work_registry_.begin(), proc_work_registry_.end());
-        std::vector<edm::DetSet<Phase2TrackerDigi>> sorted_and_merged;
-
-        edm::DetSetVector<Phase2TrackerDigi>* pr = new edm::DetSetVector<Phase2TrackerDigi>();
-
-        std::vector<Registry>::iterator it = proc_work_registry_.begin(), it2 = it + 1, end = proc_work_registry_.end();
-        while (it < end) {
-          sorted_and_merged.push_back(edm::DetSet<Phase2TrackerDigi>(it->detid));
-          std::vector<Phase2TrackerDigi>& digis = sorted_and_merged.back().data;
-          // first count how many digis we have
-          size_t len = it->length;
-          for (it2 = it + 1; (it2 != end) && (it2->detid == it->detid); ++it2) {
-            len += it2->length;
-          }
-          // reserve memory
-          digis.reserve(len);
-          // push them in
-          for (it2 = it + 0; (it2 != end) && (it2->detid == it->detid); ++it2) {
-            digis.insert(digis.end(), &proc_work_digis_[it2->index], &proc_work_digis_[it2->index + it2->length]);
-          }
-          it = it2;
-        }
+      }
+    }
 
-        edm::DetSetVector<Phase2TrackerDigi> proc_raw_dsv(sorted_and_merged, true);
-        pr->swap(proc_raw_dsv);
-        event.put(std::unique_ptr<edm::DetSetVector<Phase2TrackerDigi>>(pr), "ProcessedRaw");
-        delete buffer;
+    // store digis in edm collections
```

Comment:
Note, this was moved out of the outer most `for` loop. This avoids redoing the work again and again only to replace what was being pushed into the `edm::Event`.